src/Microsoft.AspNetCore.Mvc.Razor/Internal/PrecompiledViewsFeatureProvider.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Internal
+{
+    public class PrecompiledViewsFeatureProvider : IApplicationFeatureProvider<PrecompiledViewsFeature>
+    {
+        public static readonly string PrecompiledResourcePrefix = "__RazorPrecompiledView__.";
+        private const string DllExtension = ".dll";
+        private const string PdbExtension = ".pdb";
+
+        public void PopulateFeature(IEnumerable<ApplicationPart> parts, PrecompiledViewsFeature feature)
+        {
+            foreach (var assemblyPart in parts.OfType<AssemblyPart>())
```

Comment:
The problem isn't that we're adding a new interface, it's that they all have to be defined in `.Core`